### PR TITLE
New&beautiful

### PR DESCRIPTION
**Pull request by floppyMike**
cleaner and more flexible overhaul

### CODE REVIEW

File: SPS Communication/AuthElement.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+private:
+	std::string m_code;
+};
+
+template<typename Impl>
+class EAuthParser
+{
+	const Impl* underlying() const noexcept { return static_cast<const Impl*>(this); }
```
Comment (jegrab) [inline SPS Communication/AuthElement.h:28]: Hello there!

Comment (jegrab) [reply SPS Communication/AuthElement.h:28]: Hi

Comment (floppyMike) [reply SPS Communication/AuthElement.h:28]: nice

**Comment by floppyMike** [comment]
some unnecessary files.

**Pull request merged**